### PR TITLE
[iOS] Fix for dictation not working properly

### PR DESCRIPTION
**Pull request by Velin92**
We found a way to intercept dictation, which would not work if the content is reconciliated while the dictation is still in progress, so we delay the `reconciliateIfNeeded()` function until the dictation state is false.

### CODE REVIEW

**Comment by Velin92** [comment]
@stefanceriu <3

File: platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift [hunk 1]
```diff
@@ -172,6 +202,10 @@ public class WysiwygTextView: UITextView {
 
         wysiwygDelegate.textView(self, didReceivePasteWith: provider)
     }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
```
Comment (stefanceriu) [inline platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift:207]: I don't think this is needed

File: platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift [hunk 2]
```diff
@@ -44,6 +44,9 @@ protocol WysiwygTextViewDelegate: AnyObject {
 public protocol MentionDisplayHelper { }
 
 public class WysiwygTextView: UITextView {
+    private var lastInputMode: String?
+    private(set) var isDictationRunning = false
```
Comment (stefanceriu) [inline platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift:48]: Maybe it's better if this is property computed based on the `lastInputMode`?

File: platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift [hunk 3]
```diff
@@ -53,12 +56,39 @@ public class WysiwygTextView: UITextView {
...
     }
     
     required init?(coder: NSCoder) {
         super.init(coder: coder)
+        commonInit()
+    }
+    
+    private func commonInit() {
         contentMode = .redraw
+        NotificationCenter.default.addObserver(self,
```
Comment (stefanceriu) [inline platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift:69]: I wonder how this behaves when multiple textviews are present on screen. Should we maybe check for isFirstResponder?

File: platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift (hunk 2 above)
Comment (Velin92) [reply platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift:48]: it needs to be changed also according to two override methods for ending the recording and when the recording fails

Comment (stefanceriu) [reply platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift:48]: Maybe those should clean up the `lastInputMode` instead?

File: platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift (hunk 1 above)
Comment (Velin92) [reply platforms/ios/lib/WysiwygComposer/Sources/WysiwygComposer/Components/WysiwygComposerView/WysiwygTextView.swift]: yeah the stack overflow solution used vert old code, now these are deallocated automatically

**Review by stefanceriu — APPROVED** [review]
lgtm 👍

**Pull request merged**